### ISSUE TITLE
Two issues seen in Android 9 and above on http and connect to an access point

### ISSUE THREAD

**Issue by BillMac4440** [flutter/flutter#74399]
In 'package:http/http.dart', In Android 9 and above, http is not allowed.  I tried the work-around in the security setup in the manifest files, but that did not help.  I posted an issue on that GitHub but it was closed and was told to post it here.

In 'package:wifi_configuration/wifi_configuration.dart', there is a change in Android 10 in connecting to an access point.  You now have to give a suggested list and it says it might not use that.  Posted an issue on that GitHub.   Seems to be pull requests to fix it, but no action.

My flutter app was a test of connecting to the softAP of an embedding board and then doing an http post to give it a WiFi network to connect to.  The app works fine on Android 6 to 8, but breaks on 9 and 10.  Running out of old devices that still work!

**Comment by TahaTesser** [comment]
hI @BillMac4440 
####  For the first issue
Can you please provide your `flutter doctor -v` and `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) without using `wifi_configuration`, just `package:http/http.dart`.

#### For the second issue
Please file an issue on `wifi_configuration`'s dedicated [GitHub](https://github.com/RohitKumarMishra/wifi_configuration/issues)

Thank you

**Comment by BillMac4440** [comment]
C:\Data\FlutterApps\CM_Utilities\CM_Connection_App>flutter doctor -v

[√] Flutter (Channel dev, 1.26.0-8.0.pre, on Microsoft Windows [Version 
10.0.19042.746], locale en-US)

• Flutter version 1.26.0-8.0.pre at C:\Data\flutter

• Framework revision b9d06fffb2 (2 weeks ago), 2021-01-07 18:36:48 -0800

• Engine revision 42a8d4c681

• Dart version 2.12.0 (build 2.12.0-179.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 
30.0.3)

• Android SDK at C:\Users\bmccroskey\AppData\Local\Android\sdk

• Platform android-30, build-tools 30.0.3

• Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java

• Java version OpenJDK Runtime Environment (build 
1.8.0_242-release-1644-b01)

• All Android licenses accepted.

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 
16.8.3)

• Visual Studio at C:\Program Files (x86)\Microsoft Visual 
Studio\2019\Community

• Visual Studio Community 2019 version 16.8.30804.86

• Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)

• Android Studio at C:\Program Files\Android\Android Studio

• Flutter plugin can be installed from:

https://plugins.jetbrains.com/plugin/9212-flutter

• Dart plugin can be installed from:

https://plugins.jetbrains.com/plugin/6351-dart

• Java version OpenJDK Runtime Environment (build 
1.8.0_242-release-1644-b01)

[√] Connected device (1 available)

• Windows (desktop) • windows • windows-x64 • Microsoft Windows [Version 
10.0.19042.746]

• No issues found!

If I try to do a flutter run, it tries to build for Windows and nothing 
is supported on the channels to the libraries.

I don't have a minimal app as I need to connect to that access point to 
send the http.post.

See 
https://android-developers.googleblog.com/2018/08/introducing-android-9-pie.html 
and go to the "HTTPS by default" section of the doc.

I tried all that and no luck.

I did post an issue on wifi_configuration's GitHub.

--Bill

**Comment by TahaTesser** [comment]
Hi @BillMac4440 
To better address the issue, Can you please provide  `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
Thank you

**Comment by BillMac4440** [comment]
[image:
2021_01_22_11_07_25_flutter_app_C_Data_flutter_.pub_cache_hosted_pub.dartlang.org_http_0.12.2_lib.png]

**Comment by BillMac4440** [comment]
Ran it on an emulator with API 30 on Android 11.
This error happens from Android 9 up.

**Comment by TahaTesser** [comment]
Hi @BillMac4440 
Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by BillMac4440** [comment]
Yes, it was a change in Android, but Flutter need a way to work around it.  The post on how to fix it did not work on Flutter.  The use case is you have an IoT product and are setting it up.  As it has no config, it boots with a hot spot to connect to for configuration.  You give it the network config over that socket once you connect, then the IoT device is configured.  I've only had to do that on the stove, light switches, mesh nodes, etc.  Flutter is just a great tool to quickly set up an app for this.  Like I said, it runs great on Android 6 to 8, but new devices come up with 9 to 11 and break this.  The IoT space is really growing.  Android and Flutter along with it needs to embrace this.  I see both of these issues as walking away from the embedded world.  I am a firmware engineer on embedded code and Flutter has been the best tool for me to get something working to connect to embedded devices.  I have searched the various posts you mention, but all go back to the write-up on the Android 9 release that did not work on Flutter.  Somehow there needs to be a switch to turn this off for this simple use case to configure an IoT device using http.
